### PR TITLE
#160451570 Users should be able to follow each other

### PR DESCRIPTION
**Pull request by e0works**
#### What does this PR do?
- Implement follow functionality

#### Description of Task to be completed?
- create a list following page
- create a list followers page 
- update profile page to show following and followers
- follow button on view article page should follow and unfollow an author as the case may be

#### How should this be manually tested?
- start the app
- register two users
- create an article with the first user account
- view the article with the second user account click follow
- the first user should receive a notification, of a new follower
- go to your profile by clicking on the username on the header
    you should see you following and followers count
- click on view on the button for the followers or following 
- you should see a corresponding list page

#### Any background context you want to provide?
NA

#### What are the relevant pivotal tracker stories?
[#160451570](https://www.pivotaltracker.com/story/show/160451570, "user follow functionlaity")

#### Screenshots (if appropriate)
screen shot 2018-11-27 at 12 56 19 am
screen shot 2018-11-27 at 12 57 48 am
screen shot 2018-11-27 at 12 58 05 am
screen shot 2018-11-27 at 12 58 26 am

### CODE REVIEW

File: webpack.config.dev.js [hunk 1]
```diff
@@ -16,9 +16,14 @@ module.exports = merge(common, {
     new webpack.DefinePlugin({
       'process.env': {
         NODE_ENV: JSON.stringify('development'),
+<<<<<<< HEAD
         CLIENT_REDIRECT: JSON.stringify('http://localhost:8080'),
```
Comment (koiic) [inline webpack.config.dev.js]: Fix this config issue

File: src/components/listFollows/Followers.jsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
+// react libraries
+import React, { Fragment, Component } from 'react';
+
+// third-party libraries
+import PropTypes from 'prop-types';
+
+// modules
```
Comment (Mrshinezee) [inline src/components/listFollows/Followers.jsx]: change to 
// component

**Review by koiic — CHANGES_REQUESTED** [review]

File: src/reducers/follow/follow.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
+// modular importation
```
Comment (Mrshinezee) [inline src/reducers/follow/follow.js]: change to 
// action types

**Review by Mrshinezee — CHANGES_REQUESTED** [review]

**Review by Mrshinezee — APPROVED** [review]

**Review by koiic — APPROVED** [review]

**Review by KaiserPhemi — CHANGES_REQUESTED** [review]
@Victor-Ugwueze 
Please rebase.

**Comment by e0works** [comment]
@KaiserPhemi, I just rebased. Please review thank you

**Pull request merged**